### ISSUE TITLE
Does this work

### ISSUE THREAD

**Issue by ericwooley** [ianks/vim-tsx#1]
I tried installing it, can't seem to get it to work right. I am also using syntastic if that matters.

**Comment by ianks** [comment]
yes it does, what does your `filetype` in vim read when you open a tsx file?

**Comment by ianks** [comment]
`:set ft`

**Comment by indiesquidge** [comment]
Hmm, yeah I can't get this to work either. The dependency is on `leafgarland/typescript-vim`, correct? It's really just a copy of `mxw/vim-jsx` (with different syntax selectors), which works fine for me, including correct indentation and commenting
<img width="300" alt="screen shot 2016-06-11 at 8 30 57 pm" src="https://cloud.githubusercontent.com/assets/3409645/15988819/7df40698-3013-11e6-882e-6daec9c6706a.png">

**Comment by ianks** [comment]
the dependency is on typescript-vim. @indiesquidge does anything change when you switch the plugin ordering?

**Comment by indiesquidge** [comment]
@ianks, good question, I hadn't considered that. Unfortunately, after trying both orders, nothing changed and the plugin still won't work 😕

**Comment by ianks** [comment]
Are all your plugins up to date?

**Comment by indiesquidge** [comment]
Yeah

**Comment by indiesquidge** [comment]
This all works for you seamlessly? Commenting, indentation, highlighting? Everything just how it does in a `.jsx` file with `mxw/vim-jsx`?

**Comment by ianks** [comment]
Yes it does.

**Comment by ianks** [comment]
are you in a *.tsx file? can you paste the output of  `:set ft` in from a tsx buffer?

**Comment by ianks** [comment]
I looked in your dotfiles, this line is causing it not to work:

`autocmd BufRead,BufNewFile *.tsx setlocal filetype=typescript`

**Comment by ianks** [comment]
setting the FT is done by this plugin, and it should be `typescript.tsx`

**Comment by indiesquidge** [comment]
Ah, that's an old line that's been removed locally for a bit now. Doesn't seem to be the cause of the issue still though.

**Comment by indiesquidge** [comment]
Strangely, even without that line, my filetype is still set to `typescript`. If I manually set it it to `typescript.tsx`, some problems are solved! Syntax highlighting will show up correctly and indentation works as expected. However, commenting lines out still acts like it's a TypeScript comment (`// ...`) rather than a TSX comment (`{/* ... */}`)

**Comment by ianks** [comment]
Does that work on vim-jsx?

**Comment by indiesquidge** [comment]
The commenting feature? Yeah, it works. I'm wondering what's keeping my filetype from being set correctly now. I have nothing left in my dotfiles that explicitly sets any filetypes, so I would think the plugin's setter would work 😖

**Comment by ianks** [comment]
do you use https://github.com/Quramy/tsuquyomi ?

**Comment by indiesquidge** [comment]
Ha, yeah, that's the issue. Looks like `ianks/vim-tsx` _must_ come after `Quramy/tsuquyomi`. I'm feeling like I should bring it to `Quramy`'s attention that [his plugin is causing conflicts](https://github.com/Quramy/tsuquyomi/blob/b05558d5c97dd201386ee6ebf3835cbf76d90aff/ftdetect/typescript.vim#L1) with something that is meant to work with `.tsx` files.

**Comment by ianks** [comment]
yeah, a pr to them setting the ft to typescript.tsx should be backwards compatible, if someone were to make that pr :wink:

**Comment by indiesquidge** [comment]
Looks like `leafgarland/typescript-vim` [sets this as well](https://github.com/leafgarland/typescript-vim/blob/eaeab881826fc06ca8cf6d40ac5657edda93a69a/ftdetect/typescript.vim). Two PRs!

**Comment by indiesquidge** [comment]
Now that I think about it, if both of these libraries are explicitly setting their filetypes to be what they want, should we not just do the same for this plugin?

**Comment by indiesquidge** [comment]
Note that this is even an issue between `mxw/vim-jsx` and `pangloss/vim-javascript`.

**Comment by renebrandel** [comment]
![image](https://cloud.githubusercontent.com/assets/4989523/18056479/5cc68bbc-6e0c-11e6-9967-5279c7a5e8a3.png)
I have put this plugin after all the other named plugins. Everything works great but the brackets. Can you recreate this?

**Comment by renebrandel** [comment]
It also doesn't check for quote escapes within the code.
![image](https://cloud.githubusercontent.com/assets/4989523/18056645/0ce36fba-6e0d-11e6-9f09-ea6161a83038.png)

**Comment by ianks** [comment]
@brene I think it actually has to do with arrow functions:

![arrow-funcs-bug](https://s3-us-west-1.amazonaws.com/adhawk-random-public/vim-tsx-bug-anonymous-funcs.png)

I do not have to time right now to try to debug this, however.

**Comment by lpender** [comment]
It definitely does not indent properly:

Before adding:

![image](https://user-images.githubusercontent.com/1406554/38273763-ceb8061a-375a-11e8-9645-e945eee306aa.png)

After adding:

![image](https://user-images.githubusercontent.com/1406554/38273810-f36ee12c-375a-11e8-8e78-bbcb1eeb972c.png)
